File: python/pages/login_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage
from .home_page import HomePage


class LoginPage(BasePage):

    EMAIL = (By.ID, "email")
    PASSWORD = (By.ID, "password")
    LOGIN_BTN = (By.ID, "login-btn")

    def enter_email(self, email):
        self.type(self.EMAIL, email)
        return self

    def enter_password(self, password):
        self.type(self.PASSWORD, password)
        return self

    def click_login(self):
        self.click(self.LOGIN_BTN)
        return HomePage(self.driver)

File: src/main/java/com/example/pages/LoginPage.java
package com.example.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage extends BasePage {

    private final By username = By.id("username");
    private final By password = By.id("password");
    private final By loginBtn = By.id("loginBtn");
    private final By loginMessage = By.id("loginMessage");

    private final String url = "http://localhost:8080/login.html";

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    public LoginPage load() {
        driver.get(url);
        return this;
    }

    public LoginPage enterUsername(String username) {
        type(this.username, username);
        return this;
    }

    public LoginPage enterEmail(String e) {
        // Alias for backward compatibility
        return enterUsername(e);
    }

    public LoginPage enterPassword(String p) {
        type(password, p);
        return this;
    }

    public HomePage clickLogin() {
        click(loginBtn);
        return new HomePage(driver);
    }

    public String getLoginMessage() {
        return getText(loginMessage);
    }
}
